instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/__init__.py

```diff
@@ -327,8 +365,18 @@ def _instrumented_lambda_handler_call(
                     SpanAttributes.FAAS_EXECUTION,
                     lambda_context.aws_request_id,
                 )
+                _safe_execute_hook(
+                    request_hook, span, lambda_event, lambda_context
+                )
 
-            result = call_wrapped(*args, **kwargs)
+            lambda_error = None
+            try:
+                result = call_wrapped(*args, **kwargs)
+            except Exception as err:
+                lambda_error = err
+                raise err
+            finally:
+                _safe_execute_hook(response_hook, span, result, lambda_error)
```

Comment:
i think this will fail with `UnboundLocalError: local variable 'result' referenced before assignment` in case `call_wrapped` raises an exception.

Reply:
Fixed, thanks